Userland/Libraries/LibWeb/WebAssembly/WebAssemblyTableConstructor.cpp

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2021, Ali Mohammad Pur <mpfard@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibJS/Runtime/GlobalObject.h>
+#include <LibJS/Runtime/TypedArray.h>
+#include <LibWeb/Bindings/WindowObject.h>
+#include <LibWeb/WebAssembly/WebAssemblyObject.h>
+#include <LibWeb/WebAssembly/WebAssemblyTableConstructor.h>
+#include <LibWeb/WebAssembly/WebAssemblyTableObject.h>
+#include <LibWeb/WebAssembly/WebAssemblyTablePrototype.h>
+
+namespace Web::Bindings {
+
+WebAssemblyTableConstructor::WebAssemblyTableConstructor(JS::GlobalObject& global_object)
+    : NativeFunction(*global_object.function_prototype())
+{
+}
+
+WebAssemblyTableConstructor::~WebAssemblyTableConstructor()
+{
+}
+
+JS::Value WebAssemblyTableConstructor::call()
+{
+    vm().throw_exception<JS::TypeError>(global_object(), JS::ErrorType::ConstructorWithoutNew, "WebAssembly.Table");
+    return {};
+}
+
+JS::Value WebAssemblyTableConstructor::construct(FunctionObject&)
+{
+    auto& vm = this->vm();
+    auto& global_object = this->global_object();
+
+    auto descriptor = vm.argument(0).to_object(global_object);
+    if (vm.exception())
+        return {};
+
+    auto element_value = descriptor->get_without_side_effects("element");
```

Comment:
`get_without_side_effects` is something you rarely need in general and pretty much never need in any "user-facing" code. As implemented, this wouldn't work with either Proxies or accessors. The spec is *really* unclear (`descriptor["element"]`? `:yakfused:`), but based on Firefox's implementation and a general git feeling this should be a regular `get()` + exception check :)

Reply:
I don't think it makes much sense for this to be anything except a string, but I'll go with your suggestion.

Reply:
> I don't think it makes much sense for this to be anything except a string

Sure, but that's not what I'm saying! This is about `new WebAssembly.Table({ get element() { ... } })` not working, for instance.